master/cmd/determined-master/root.go

```diff
@@ -137,6 +143,35 @@ func getConfig(configMap map[string]interface{}) (*internal.Config, error) {
 	return config, nil
 }
 
+func applyBindMount(configMap map[string]interface{}) (map[string]interface{}, error) {
+	bindMountField, fieldExists := configMap["auto_bind_mount"]
+	if !fieldExists {
+		return configMap, nil
+	}
+	vBindMount, ok := bindMountField.(string)
+	if !ok {
+		return nil, errors.New("wrong type for bind mount field")
+	}
+
+	if vBindMount != "no" {
+		_, tcdExisted := configMap["task_container_defaults"]
+		if !tcdExisted {
+			newTCD := make(map[string]interface{})
+			bindMounts := make([]model.BindMount, 1)
+			bindMounts[0] = model.BindMount{
+				HostPath:      vBindMount,
+				ContainerPath: "/run/determined/shared_fs",
+			}
+			newTCD["bind_mounts"] = bindMounts
+			configMap["task_container_defaults"] = []map[string]interface{}{newTCD}
```

Comment:
if a user has a `--master-config-path`, having this option enabled will wipe out `task-container-defaults` section. 
should we disallow having both `--master-config-path` and this new option together (probably on the CLI side)? or document this behavior?

Reply:
Added code to handle the case when user specified a `--master-config-path`.